awx/settings/defaults.py

```diff
@@ -882,7 +882,7 @@
         'awx.main.access': {'level': 'INFO'},  # very verbose debug-level logs
         'awx.main.signals': {'level': 'INFO'},  # very verbose debug-level logs
         'awx.api.permissions': {'level': 'INFO'},  # very verbose debug-level logs
-        'awx.analytics': {'handlers': ['external_logger'], 'level': 'INFO', 'propagate': False},
+        'awx.analytics': {'handlers': ['external_logger', 'activity_stream'], 'level': 'INFO', 'propagate': False},
```

Comment:
I don't think this is going to log what you think it will. This encompasses activity stream and other stuff like job events that you really don't want in the file. Perhaps you would do good to add a new line that does

```python
        'awx.analytics.activity_stream': {'handlers': ['activity_stream']},
```

in order to capture logs sent to here:

https://github.com/ansible/awx/blob/1a56272eaf74f34ba1ee538de1b41d9d6a8278a6/awx/main/signals.py#L73

Good chance that will capture the logs. But we shouldn't take it on faith that that is what you want either. You should do some stuff and then open up the `activity_stream.log` file and see what actually goes in. I highly suspect that the `extra=` sent to the logger will be ignored, and you will get an uninteresting message.

Assuming this is where you land with this all, and that your intent really is to get the _changes_, then your option will be to deal with the python logging formatter. Hopefully that can be done without an additional class, and it will be fairly clean to wrap up.

Reply:
@AlanCoding I agree with the change to select only the awx.analytics.activity_stream events. I overlooked that when setting the logger.
Regarding the outcome on the log file, the current output is a JSON value that contains the important data about the change made, the actor, and the extras.
We may instead process the output to match other logs' format and add the extras because the default logger is not verbose enough for the purpose of troubleshooting.

Reply:
@AlanCoding Based on your feedback I made additional changes.